src/routes/demographics/Content.svelte

```diff
@@ -143,21 +166,61 @@
           {#if schema[question].type === 'text'}
             <div
               class="mzp-c-choice mzp-c-choice-text"
-              class:mzp-is-error={!$zipValidity.valid}>
+              class:mzp-is-error={
+                inputFormatters.showErrors(question) && // only show errors if explicitly told, e.g. on blur
+                inputFormatters.hasValidator(question) && // only show errors if there's even a validator
+                inputFormatters[question].isInvalid(results[question] // only show errors if the input is invalid
+            )}>
               <input
-                use:validateZip={results[question]}
                 type="text"
-                bind:value={results[question]} />
+                class:right={inputFormatters[question].alignRight}
+                on:blur={(event) => {
+                  if (inputFormatters[question] && inputFormatters[question].blur) {
+                    results[question] = inputFormatters[question].blur(event);
+                  }
+                }}
+                on:focus={(event) => {
+                  if (inputFormatters[question] && inputFormatters[question].focus) {
+                    results[question] = inputFormatters[question].focus(event);
+                  }
+                }}
+                on:input={async (event) => {
+                  let value;
+                  if (inputFormatters[question] && inputFormatters[question].input) {
+                    value = inputFormatters[question].input(event);
+                    event.target.value = value;
+                  } else {
+                    value = event.target.value;
+                  }
+                  
+                  results[question] = value;
+                  await tick();
+                }}
+                value={results[question]}
+                />
+                <span style="min-height: 24px; display: block;">
+              {#if 
+                  inputFormatters.showErrors(question) && // show errors if blurred
+                  inputFormatters.hasValidator(question) && // show errors if there's a validator for this question
+                  inputFormatters[question].isInvalid(results[question])
+                }
+                <span
+                  class="mzp-c-fieldnote"
+                  transition:fly={{ duration: 300, y: 5 }}>
+                  {inputFormatters[question].isInvalid(results[question])}
+                </span>
+              {/if}
+              </span>
               <!-- show validation errors here -->
-              <span style="min-height: 24px; display: block;">
+              <!-- <span style="min-height: 24px; display: block;">
```

Comment:
Is there a reason to comment this out vs. just remove it? It's in the git history if you want it back... if you plan to do something with it like re-enable it when an issue is fixed please add a comment :)

Reply:
No good reason, removing now.